pkg/ebpf/bpf_client.go

```diff
@@ -659,11 +656,11 @@ func (l *bpfClient) AttacheBPFProbes(pod types.NamespacedName, podIdentifier str
 
 	// Two go routines can try to attach the probes at the same time
 	// Locking will help updating all the datastructures correctly
-	value, _ := l.AttachProbesToPodLock.LoadOrStore(podIdentifier, &sync.Mutex{})
-	attachProbesLock := value.(*sync.Mutex)
-	attachProbesLock.Lock()
-	log().Debugf("Got the attachProbesLock for Pod: %s, Namespace: %s, PodIdentifier: %s", pod.Name, pod.Namespace, podIdentifier)
-	defer attachProbesLock.Unlock()
+	value, _ := l.PodIdentifierLock.LoadOrStore(podIdentifier, &sync.Mutex{})
```

Comment:
can we make this `PodIdentifierLock` private such that bpfclient is only aware of this lock?

we can remove the `GetPodIdentifierLockMap()` from bpf client interface. this will simplify `rpchandler Delete` method and push down this lock/unlock to bpf client.

As `bpfclient` is the owner of making changes related to attach/delete probe. it can handle this locking functionality internally and not leave it on caller to properly lock/unlock it which is what we are currently doing in Delete routine.